Cython/Compiler/ExprNodes.py

```diff
@@ -4749,9 +4749,29 @@ def analyse_types(self, env, getting=True):
                 analyse_base=False)
 
         if self.start:
-            self.start = self.start.analyse_types(env)
+            self.start = CondExprNode(
+                self.start.pos,
+                true_val = IntNode(self.start.pos, value = '0'),
+                false_val = self.start,
+                test = PrimaryCmpNode(
+                    self.start.pos,
+                    operand1 = self.start,
+                    operator = 'is',
+                    operand2 = NoneNode(self.start.pos),
+                )
+            ).analyse_types(env)
         if self.stop:
-            self.stop = self.stop.analyse_types(env)
+            self.stop = CondExprNode(
+                self.stop.pos,
+                true_val = IntNode(self.stop.pos, value = 'PY_SSIZE_T_MAX'),
```

Comment:
This C macro constant needs to be made a C integer explicitly, i.e. add `type=PyrexTypes.c_py_ssize_t_type`. That's also the main reason why travis is unhappy.
The start `IntNode` should get its type automatically.

Reply:
Declaring the `type` explicitly in the constructor worked initially, but the subsequent call to `CondExprNode.analyse_types()` appears to switch the `IntNode`'s `type` back to `py_object_type`. I ultimately ended up adding a call to `coerce_type(PyrexTypes.c_py_ssize_t_type, env)` following the call to `analyse_types()`, which allowed the `type` to persist inside the `IntNode`.